deb/package.go

```diff
@@ -321,24 +338,56 @@ func (p *Package) MatchesArchitecture(arch string) bool {
 	return p.Architecture == arch
 }
 
+// providesDependency checks if the package `Provide:`s the dependency, assuming that the architecture matches.
+// If the `Provides:` entry includes a version number, it will be considered when checking the dependency.
+func (p *Package) providesDependency(dep Dependency) (bool, error) {
+	for _, provided := range p.Provides {
+		providedDep, err := ParseDependency(provided)
+		if err != nil {
+			return false, err
+		}
+		// The only relation allowed here is `=`.
+		// > The relations allowed are [...]. The exception is the Provides field, for which only = is allowed.
+		// > [...]
+		// > A Provides field may contain version numbers, and such a version number will be considered when
+		// > considering a dependency on or conflict with the virtual package name.
+		// -- https://www.debian.org/doc/debian-policy/ch-relationships.html
+		switch providedDep.Relation {
+		case VersionDontCare:
+			if providedDep.Pkg == dep.Pkg {
+				return true, nil
+			}
+		case VersionEqual:
+			providedVersion := providedDep.Version
+			if providedDep.Pkg == dep.Pkg && versionSatisfiesDependency(providedVersion, dep) {
+				return true, nil
+			}
+		default:
+			return false, fmt.Errorf("unsupported relation in Provides: %s", providedDep.String())
+		}
+	}
+	return false, nil
+}
+
 // MatchesDependency checks whether package matches specified dependency
 func (p *Package) MatchesDependency(dep Dependency) bool {
 	if dep.Architecture != "" && !p.MatchesArchitecture(dep.Architecture) {
 		return false
 	}
 
-	if dep.Relation == VersionDontCare {
-		if utils.StrSliceHasItem(p.Provides, dep.Pkg) {
-			return true
-		}
-		return dep.Pkg == p.Name
+	if providesDep, _ := p.providesDependency(dep); providesDep { // Is ignoring errors the right thing to do here?
```

Comment:
should probably be logged as warning...

Reply:
OK, I added a log message there.

I wasn't sure if I should use `"github.com/rs/zerolog/log"` or `"log"` (from the standard library) as both seem to be used in Aptly. I went with the former.